bde/ml/models.py

```diff
@@ -320,70 +324,103 @@ def __init__(
         self.seed = seed
 
         self.params_: Union[FrozenDict, Dict] = dict()
-        self.history_: Optional[Array] = None
-        self.model_: Optional[BasicModule] = None
+        self.history_: Array = None  # type: ignore
+        self.model_: BasicModule = None  # type: ignore
         self.is_fitted_: bool = False
         self.n_features_in_: Optional[int] = None
 
-    def tree_flatten(self) -> Tuple[Sequence[ArrayLike], Any]:
+    def tree_flatten(self) -> Tuple[Sequence[Union[ArrayLike, Dict]], Any]:
         r"""Specify how to serialize estimator into a JAX pytree.
 
         Returns
         -------
         Tuple[Sequence[ArrayLike], Any]
             A tuple with 2 elements:
-             - The `children`, containing arrays & pytrees (2 elements).
-             - The `aux_data`, containing static and hashable data (13 elements).
+             - The `children`, containing arrays & pytrees.
+             - The `aux_data`, containing static and hashable data.
         """
+        atts = {
+            "batch_size": self.batch_size,
+            "validation_size": self.validation_size,
+            "seed": self.seed,
+            "is_fitted_": self.is_fitted_,
+            "n_features_in_": self.n_features_in_,
+        }
         children = (
             self.model_,
```

Comment:
Not saving the posterior samples?

Reply:
The `FullyConnectedEstimator` class is a regular neural network with no sampling. The `BDEEstinator` class implements the sampling functionality and it does save the samples when serializing (line 785).